Cargo.toml

```diff
@@ -55,6 +55,8 @@ needless_doctest_main = "allow"
 # see #878
 result_large_err = "allow"
 large_enum_variant = "allow"
+cognitive_complexity = "allow"
```

Comment:
how often was this showing up? If it's only one or two place I might  prefer allowing it an the individual functions since IMO it's nice that clippy calls out complex functions even if we decide against refactoring.

Reply:
There are a few places now. I will annotate them individually.

Reply:
Done.